package cms_8120;
import "type/cmd.proto";
option java_package = "com.hikvision.finance.fms.cms.hpp.center";
option java_outer_classname = "RegionProto";
option java_generic_services = true;


//组织表
message Region
{
    //必须字段
    optional int32 id                       = 1;    // 在数据库中的流水号
    optional int32 nParentId                = 2;    // 上级组织(父区域)ID
    optional string strName                 = 3;    // 组织名称
    optional string strCode                 = 4;    // 组织代码
    optional string strInterAreaCode        = 5;    // 组织树节点号
    optional string strPath                 = 6;    // 组织树节点路径
    optional int32 nSortId                  = 7;    // 组织的序号（从1开始按照添加顺序依次添加）
    optional string strTxt                  = 8;    // 其他信息
}

message ReqRegion
{
    optional int32 cmd                	    = 1;    // CMD_DT_REGION_LIST
    optional int32 id                       = 2;    // 在数据库中的流水号
	optional int32 nParentId                = 3;    // 上级组织(父区域)ID
    optional string strName                 = 4;    // 组织名称
    optional string strCode                 = 5;    // 组织代码
    optional string strInterAreaCode        = 6;    // 组织树节点号
    optional string strPath                 = 7;    // 组织树节点路径
    optional int32 nSortId                  = 8;    // 组织的序号（从1开始按照添加顺序依次添加）
    optional string strTxt                  = 9;    // 其他信息
}

message RegionList
{
    optional int32 cmd                      = 1;    // CMD_DT_REGION_LIST
    repeated Region region                  = 2;    // 组织列表
}


service ModifyDeviceService {
	
	rpc QueryRegionInfo(ReqRegion) returns (RegionList);			// CMD_DT_REGION_INFO(获取组织信息)
	
	rpc QueryAllDvrInfo(ReqRegion) returns (RegionList);				// CMD_DT_REGION_LIST(获取所有组织信息)
	
}

